stackslib/src/chainstate/nakamoto/test_signers.rs

```diff
@@ -147,14 +252,66 @@ impl TestSigners {
         let signature = sig_aggregator
             .sign(msg.as_slice(), &nonces, &sig_shares, &key_ids)
             .expect("aggregator sig failed");
+        ThresholdSignature(signature)
+    }
 
-        test_debug!(
-            "Signed Nakamoto block {} with {} (rc {})",
-            block.block_id(),
-            &self.aggregate_public_key,
-            cycle
-        );
-        block.header.signer_signature = ThresholdSignature(signature);
+    /// Reorder a list of signatures to match the order of the reward set.
+    pub fn reorder_signatures(
```

Comment:
I'm rather confused as to why this method and its caller (`sign_block_with_reward_set`) need to exist?  If a block is signed by iterating through the reward set's private keys, then it would always be the case that the signatures are in order.  No one should be trying to check a block's signatures against a different reward set than the one that signed it, so `sign_block_with_reward_set` as written seems buggy.  If the test wants to sign a block with a different reward set than the one configured in this `TestSigners` instance, then perhaps the test should create a `TestSigners` for that reward set?

Reply:
Yeah I understand the confusion. For one, this function isn't signing the block using the reward set's keys (the `RewardSet` also doesn't have private keys), it's signing the block using `TestSigners`'s keys and _using_ the reward set to separately:

1. (most importantly) not sign the block with any TestSigner keys that aren't in the reward set
2. reorder signatures to match the reward set

For a use case, imagine a testing scenario like this:

- Set up your test with `TestSigners(alice, bob)`
- Next, stack with just bob
- Sign some blocks
- Later stack with alice
- Sign some blocks

In that cash, `TestSigners` would appropriately only sign blocks using the correct signers at that time. You could do this by manually changing `test_signers.signer_keys`, too, but `TestSigners` is generally just used for one-time test setup and then passed around.

Reply:
I refactored this method to do the (for each key in reward_set, sign if TestSigner has that key) approach. LMK if you think this needs more work